soccer/gameplay/tactics/wall.py

```diff
@@ -0,0 +1,65 @@
+import composite_behavior
+import behavior
+import constants
+import robocup
+import main
+import enum
+import math
+import skills.move
+
+# This tactic builds a wall a certain distance from point A, blocking point B.
+class Wall(composite_behavior.CompositeBehavior):
+    class State(enum.Enum):
+        defense_wall = 1
+
+    def __init__(self,
+                 num_defenders = 3,                         # number of defenders we're making the wall with (default 3)
+                 curvature = 0,                             # curvature (in radians) of the wall 0 - pi/2
+                 mark_point = None,                         # what point we are defending against (default is ball)
+                 defender_point = robocup.Point(0, 0),      # what point we are defending (default is goal)
+                 defender_spacing = 3,                      # number of robot radii between the centers of the defenders in the wall
+                 dist_from_mark = 1):                       # distance from the mark point we want to build the wall
+        super().__init__(continuous=True)
+
+        self.number_of_defenders = num_defenders
+        self.curvature = curvature
+        self.mark_point = main.ball().pos if mark_point == None else mark_point
```

Comment:
On another note, this only builds a wall at the ball location at the ball location when you create the subbehavior. It doesn't update as the ball moves. Is that the wanted behavior? @Alan-Penkar

Reply:
In my mind, the defense tactic above the wall tactic will be updating the mark_point every-time it iterates, which will then force the wall to adapt. Does that seem correct or no?